haystack/nodes/preprocessor/merger.py

```diff
@@ -0,0 +1,301 @@
+
+        if window_size == 1:
+            logging.info(
+                "DocumentMerger with 'window_size=1' does nothing to the incoming documents list. "
+                "Consider removing this node or changing the value of this parameter. "
+                "If you want to merge all incoming documents into a single one, use 'window_size=0'."
+            )
+
+        if window_size:
+            if window_overlap < 0 or not isinstance(window_overlap, int):
+                raise ValueError("window_overlap must be an integer >= 0")
+
+            if window_overlap >= window_size:
+                raise ValueError("window_size must be larger than window_overlap")
+
+    def run(  # type: ignore
+        self,
+        documents: List[Document],
+        separator: Optional[str] = None,
+        window_size: Optional[int] = None,
+        window_overlap: Optional[int] = None,
+        realign_headlines: Optional[bool] = None,
+        retain_page_number: Optional[bool] = None,
+    ):
+        """
+        Merges all or a subset of the documents into bigger documents.
+
+        Will retain all metadata that is shared across all documents with the same value
+        (for example, filename might be retained of all documents come from the same file),
+
+        Can treat some special fields separately:
+        - `headlines`: the content of their `start_idx` field will be updated to reflect the actual
+            position in the merged document
+        - `page`: will be set to the smallest value found across the documents to merge.
+
+        :param separator: The separator that appears between subsequent merged documents.
+        :param window_size: how many documents to include in each merged batch. For example, if set to 2,
+                            the documents will be merged in pairs. Set to 0 to merge all documents in one
+                            single document.
+        :param window_overlap: to apply a sliding window approach over the documents groups. For example,
+                               if window_size=3 and window_overlap=2, the resulting documents will come
+                               from the merge of the following groups: `[doc1, doc2, doc3]`, `[doc2, doc3, doc4]`, ...
+        :param realign_headlines: whether to update the value of `start_idx` for the document's headlines, if found
+                                  in the metadata. Setting it to False will drop all the headline information found.
+        :param retain_page_number: whether to set the page number to the lowest value in case of mismatch across the
+                                   merged documents. Setting it to False will always drop the page number from the
+                                   merged document.
+        """
+        if not all(doc.content_type == "text" for doc in documents):
+            raise ValueError(
+                "Some of the documents provided are non-textual. Document Merger only works on textual documents."
```

Comment:
how about putting the action first:
Make sure the documents you provide contain text only because DocumentMerger only works on textual documents.

Reply:
that's because: Time to value